### ISSUE TITLE
To restart a visualization from a previously saved image.

### ISSUE THREAD

**Issue by tul-urte** [tensorflow/lucid#134]
There appears to be support to continue a visualization from a previously saved image via **param/spatial.py** which has:

`def pixel_image(shape, sd=None, init_val=None):`

However, this is not currently reachable via **param/images.py**.

It nearly works perfectly if I modify **param/images.py** (passing **init_val**) as follows:

```
def image(w, h=None, batch=None, sd=None, decorrelate=True, fft=True, alpha=False, init_val=None):
    h = h or w
    batch = batch or 1
    channels = 4 if alpha else 3
    shape = [batch, w, h, channels]
    
    if init_val is not None:
        t = tf.Variable( init_val )
        rgb = to_valid_rgb( t[..., :3], decorrelate=True, sigmoid=True )
    else:
        t = fft_image( shape, sd=sd ) if fft else pixel_image( shape, sd=sd )
        rgb = to_valid_rgb( t[..., :3], decorrelate=decorrelate, sigmoid=True )
 
    if alpha:
        a = tf.nn.sigmoid(t[..., 3:])
        return tf.concat([rgb, a], -1)
    return rgb
```

Then I can construct **param_f** as follows:

```
with Image.open( img_path ) as im:
    im_1 = np.array( im )
    
# add dummy batch dimension
im_2 = np.expand_dims( im_1, axis=0 )

# reduce less than one
init_val = im_2.astype( np.float32 ) / 256

param_f = lambda:image( scale, fft=False, init_val=init_val )
```

As I said, it nearly works perfectly, but, if I compare the results of running 
[ 64, 128, 256 ] and then restart from the 256 image through [ 384, 512]:
![image](https://user-images.githubusercontent.com/22941839/51687183-ef0da280-1fe9-11e9-8a3a-e9cc09b0e335.png)
![image](https://user-images.githubusercontent.com/22941839/51691257-5cbdcc80-1ff2-11e9-8cbb-3bfecdfdd618.png)

with a straight run through:
[ 64, 128, 256, 384, 512 ] 
![image](https://user-images.githubusercontent.com/22941839/51687231-064c9000-1fea-11e9-9b46-425c0fb4ffcf.png)
![image](https://user-images.githubusercontent.com/22941839/51691299-6cd5ac00-1ff2-11e9-8fa8-328f576cfc85.png)

You can see the difference!

I've tried this again and again across many graphs/layers/indexes and there is always something missing in the "picked-up" images.

I've already hacked my way this far and I would appreciate any help.

NB: I tried the other 3 combinations of the arguments for:

`to_valid_rgb( t[..., :3], decorrelate=True, sigmoid=True )`

but any other choices produce completely different images.

**Comment by ludwigschubert** [comment]
You raise a good point: setting a start image is a reasonable ask of a visualization framework. Unfortunately complex parameterizations like the i2ddfft one are non-trivial to initialize. (You’d either need to symbolically solve for the Fourier coefficients, or even do an optimization step in which you optimize the parameterization parameters to produce your desired start image.)

For the case of pixel backed parameterizations your request should be easy to implement, though, as you can see from our unfinished API for this. 

I’ll investigate the issue you’re seeing and hope to come back to this within a week. 

Until then, one potential source for the inconsistency you see may be the running averages in the Adam optimizer we use by default. Could you try setting it to normal SGD and see if you still get these results?
